src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -27,9 +50,44 @@ public ListPage()
 
     protected override void OnNavigatedTo(NavigationEventArgs e)
     {
+        LoadedState = ViewModelLoadedState.Loading;
         if (e.Parameter is ListViewModel lvm)
         {
-            ViewModel = lvm;
+            if (!lvm.IsInitialized
+                && lvm.InitializeCommand != null)
+            {
+                ViewModel = null;
+                lvm.InitializeCommand.Execute(null);
```

Comment:
for my own notes: 

This starts ListViewModel:InitializeAsync _on the UI thread_, then starts a BG thread to wait for the result. 

That ends up resulting in a hot second of stutter where we wait to actually fetch the items, _before_ we switch to the new page.